core/src/test/java/io/questdb/test/griffin/SqlParserTest.java

```diff
@@ -1415,6 +1415,20 @@ public void testBracedArithmeticsInsideCast() throws SqlException {
         );
     }
 
+    @Test
+    public void testCTEWithTimestampClause() throws Exception {
+        assertQuery(
+                "select-choose ts from (" +
+                        "select-choose [ts] ts from (" +
+                        "select-choose [ts] ts from (" +
+                        "select-choose [ts] ts from (select [ts] from tango)) samba timestamp (ts)" +
+                        ") rhumba" +
+                        ") chacha",
+                "WITH samba AS (tango), rhumba AS (samba TIMESTAMP(ts)), chacha AS (rhumba), unused AS (chacha) SELECT * FROM chacha",
```

Comment:
`rhumba AS (samba TIMESTAMP(ts))` - is it valid syntax? What is it supposed to select?